Dockerfile

```diff
@@ -25,8 +25,18 @@ COPY --from=builder /src/webapp /app
 # through `envsubst` by the nginx docker image entry point.
 COPY /docker/nginx-templates/* /etc/nginx/templates/
 
+# Tell nginx to put its pidfile elsewhere, so it can run as non-root
+RUN sed -i -e 's,/var/run/nginx.pid,/tmp/nginx.pid,' /etc/nginx/nginx.conf
+
+# nginx user must own the cache and etc directory to write cache and tweak the nginx config
+RUN chown -R nginx:0 /var/cache/nginx /etc/nginx
+RUN chmod -R g+w /var/cache/nginx /etc/nginx
```

Comment:
Rather than having to maintain two tmp file systems (`/tmp` and `/var/cache`), why not move both to the same place, e.g. `/tmp`?

(Sadly it seems impossible to prevent the creation of a PID file at all.)

Reply:
Ah, and by the way rather than modifying the config via a scary regexp you could just launch nginx via a parameter override:

`nginx -g "pid /tmp/nginx.pid;"`

Reply:
Contributions welcome!

Reply:
I tried to contribute under terms of AGPL-3.0, but I won't sign CLAs.